### ISSUE TITLE
plugin-taskbar: undesirable highlighting of buttons

### ISSUE THREAD

**Issue by pmattern** [lxde/lxqt#971]
The highlighting of buttons in plugin-taskbar that's normally seen upon hovering over the buttons only
![lxqt-panel_plugin-taskbar-highlight-move-other-desktop](https://cloud.githubusercontent.com/assets/9860801/13554584/0aae7942-e3ac-11e5-984b-994fb0e2153f.png)
takes place due to two actions as well where it actually does not make sense.

The first one is leaving buttons' context menu without performing an action.
E. g. just open a button's context menu and click somewhere else on the desktop to close the menu without taking any action. It doesn't matter whether or not another submenu of the context menu had been opened before closing the context menu altogether.

The second action is moving a window to another virtual desktop from the context menu of buttons in plugin-taskbar.
Steps to reproduce:
* Open several windows of a single application instance on a particular virtual desktop, e. g. some browser windows which all belong to a single instance.
* Move one or more of these windows to an arbitrary other virtual desktop by selecting "To Desktop" from the context menu of the window's button in plugin-taskbar.
* Switch to one of those other virtual desktops by clicking their button in plugin-desktopswitch or using an appropriate keyboard shortcut. The button of the window in plugin-taskbar will be highlighted.

The highlighting can be stopped by clicking or hovering over the button only. E. g. it persists after switching to an other desktop and returning to the one in question.
The second finding can not be seen when windows get moved by keyboard shortcuts or by actions from window manager's menus such as "Send to desktop" in the menu representing the regular menu of the icon in windows' upper left corner and context menu of the titlebar under Openbox.
The window manager does not seem to be involved as the findings are the same with Openbox, KWin and Xfwm.
Problem is eventually related to #681 / lxde/lxqt-panel#252 somehow.

Seen with recent VCS checkouts of LXQt on Arch Linux and LXQt 0.10 on Debian stretch (testing).

**Comment by tsujan** [comment]
This is a Qt5 bug. I've included a workaround for it in Kvantum (https://github.com/tsujan/Kvantum/commit/55bd599d26e0e5e2ccb40616cb5fcd2cd855ccc8).

https://bugreports.qt.io/browse/QTBUG-36862 can be about the same bug. They've closed it but apparently it isn't fixed yet.

**Comment by palinek** [comment]
Yes, I also believe that this is an issue in QT and already submitted bugs for something like that:
- https://bugreports.qt.io/browse/QTBUG-49222 -> is fixed, but delivered in 5.6.1 release
- and https://bugreports.qt.io/browse/QTBUG-51573 -> which has a fix already

Not: I believe that this has the very same reason as issue with highlighted plugins after menu spawn/close in lxde/lxqt#706

**Comment by tsujan** [comment]
@palinek Thanks for the links.
The Qt5 menu-hover bugs are somehow related to each other. I hope all of them will be fixed soon because they're so annoying but QTBUG-36862 seems to be forgotten.

**Comment by pmattern** [comment]
There's another similar problem, probably due to the same reason.
Several plugins remain highlighted when a component belonging to the plugin was launched by clicking on the plugin and closed by clicking somewhere outside. At least among other this is affecting plugin-[world]clock and plugin-mainmenu as seen in this screenshot
![lxqt-panel_plugin-remain-highlighted](https://cloud.githubusercontent.com/assets/9860801/15451186/fa4dc062-1fb5-11e6-87cd-cf7f4469e176.png)
taken after the menu was opened by clicking on the plugin itself and then closed by clicking somewhere else on the desktop.

**Comment by tsujan** [comment]
> There's another similar problem, probably due to the same reason.

Yes, the reason should be the same Qt bug.
